### PR TITLE
Issue/81 deploy all consumers

### PR DESCRIPTION
**Pull request by davemeyer**
Closes: #81 

deploy all consumer in the prod kube cluster

### CODE REVIEW

File: consumers/infrastructure/stream-consumer/values.yaml [hunk 1]
```diff
@@ -50,3 +51,23 @@ nodeSelector: {}
...
+    name: processmetrics
+  - topic: botany_activity
+    handler: triggerActivityNudges
+    name: triggeractivitynudges
+  - topic: botany_nudges
+    handler: sendNudges
+    name: sendnudges
+  - topic: slack_interaction
+    handler: respondToSlack
+    name: respondtosalck
```
Comment (andycmaj) [inline consumers/infrastructure/stream-consumer/values.yaml]: sp. `respondtosALck`

Comment (andycmaj) [reply consumers/infrastructure/stream-consumer/values.yaml]: future improvement: `lower` chart function
https://github.com/helm/helm/blob/master/docs/charts_tips_and_tricks.md#know-your-template-functions

```
value: {{ include "mytpl" . | lower }}
```

File: consumers/src/handlers/sendNudges/index.ts [hunk 2]
```diff
@@ -36,11 +36,10 @@ const handler: TopicMessageHandler<Nudge> = async (nudge: Nudge) => {
   })) as SlackImResponse;
 
   debug('consumers:sendNudges', 'imResponse', imResponse);
-
   try {
     await slack.chat.postMessage({
       ...nudge.message,
-      text: null,
+      text: 'hello',
```
Comment (andycmaj) [inline consumers/src/handlers/sendNudges/index.ts]: `TODO: https://github.com/asynchronous-dev/react-chat-renderer/issues/10` ?

File: consumers/infrastructure/stream-consumer/templates/deployment.yaml [hunk 3]
```diff
@@ -22,24 +22,24 @@ spec:
...
+        - name: {{ $.Chart.Name }}-{{ .handler | lower }}
+          image: "{{ $.Values.image.repository }}:{{ $.Values.image.tag }}"
+          imagePullPolicy: {{ $.Values.image.pullPolicy }}
+          args: ["--topic", "{{ .topic }}", "--handler", "{{ .handler }}"]
           env:
-            - name: TOPIC
-              value: {{ .Values.consumer.topic }}
-            - name: CONSUMER_GROUP_ID
-              value: {{ .Values.consumer.groupId }}
+            - name: CONSUMER_GROUP_ID-{{ .topic}}
```
Comment (andycmaj) [inline consumers/infrastructure/stream-consumer/templates/deployment.yaml]: just noticed this change, my bad for missing it... 

I think it might make more sense to vary the VALUE of `CONSUMER_GROUP_ID` by `.handler`. 
The way you have it now, you're changing the ENV var name, so the [code that reads it](https://github.com/asynchronous-dev/botany/blob/0b9a4b6ddf5686303a65c116524f3325e824cd87/consumers/src/config.ts#L26) will never actually read this env var.

Quick summary of how the consumer IDs are used... 
https://dzone.com/articles/understanding-kafka-consumer-groups-and-consumer-l

so basically all of our consumer handlers should be in separate consumer groups. otherwise, they might rebalance weirdly and just get kinda confusing... especially because we have multiple handlers consuming from same topic, eg. `processMetrics` could move the offset without `persistActivity` ever seeing a given message.

https://confluent.cloud/environments/t4678/clusters/lkc-4xvdz/consumerGroups/dev_consumer
ScreenCapture at Sun Sep 29 08:54:40 PDT 2019
